### PR TITLE
Downgrade react and react-dom to 19.0.0-rc.0

### PR DESCRIPTION
**Pull request by karlhorky**
### What?

Fix potential production compatibility problems with reproduction templates

With the possibility of introducing `npx` problems (like what @samcx [fixed](https://github.com/vercel/next.js/pull/66301)) or other dev problems (like what [I reported](https://github.com/vercel/next.js/issues/67783))

### Why?

According to @eps1lon, versions other than `19.0.0-rc.0` can break in production

### How?

Downgrade `react` and `react-dom` in App Router reproduction template to `19.0.0-rc.0`

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: fd6f63054f61a416ea6b2b70a4e1aad32c26bd31

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

File: examples/reproduction-template/package.json [hunk 1]
```diff
@@ -7,8 +7,8 @@
   },
   "dependencies": {
     "next": "canary",
-    "react": "19.0.0-rc-6f23540c7d-20240528",
-    "react-dom": "19.0.0-rc-6f23540c7d-20240528"
+    "react": "19.0.0-rc.0",
+    "react-dom": "19.0.0-rc.0"
```
Comment (eps1lon) [inline examples/reproduction-template/package.json:11]: Follow-up for me: Make sure the react-sync script automatically updates this

**Review by eps1lon — APPROVED** [review]
Thanks

**Comment by karlhorky** [comment]
Thanks to you too, for letting me know more about supported `react` and `react-dom` versions!

**Comment by karlhorky** [comment]
@eps1lon @samcx thanks for the review and merge

**Pull request merged**